modules/java_api/src/main/cpp/jni_common.hpp

```diff
@@ -202,3 +202,33 @@ static std::vector<size_t> jintArrayToVector(JNIEnv *env, jintArray dims)
 
     return std::vector<size_t>();
 }
+
+template<typename T>
+static jobject vectorToJavaList(JNIEnv *env, std::vector<T> items)
+{
+    static const char method_name[] = "vectorToJavaList";
+    try
+    {
+        jclass arrayClass = env->FindClass("java/util/ArrayList");
+        jmethodID arrayInit = env->GetMethodID(arrayClass, "<init>", "()V");
+        jobject arrayObj = env->NewObject(arrayClass, arrayInit);
+        jmethodID arrayAdd = env->GetMethodID(arrayClass, "add", "(Ljava/lang/Object;)Z");
+
+        for (const auto& item : items) {
+            jstring string = env->NewStringUTF(item.c_str());
```

Comment:
There would be an error if item is not `std::string`

Reply:
[Updated](https://github.com/openvinotoolkit/openvino_contrib/pull/709/commits/98c3e6debff015a71afd196771698d99b048278b#diff-7eea2d8fe6a6c9d40a378141adc603ea0b429492bd1410ae550edd7cd2d31807) to accept `std::string` items.